File: scripts/test_sentry.ts
#!/usr/bin/env node
/**
 * Standalone script to test Sentry integration from TypeScript/Node.js
 */
import * as dotenv from 'dotenv';
import * as path from 'path';

// Load environment variables
dotenv.config({ path: path.join(__dirname, '..', '.env') });

async function main() {
    try {
        // Dynamic import to ensure environment is loaded first
        const { testSentryIntegration } = await import('../sentry/ts/sentry.js');
        testSentryIntegration();
    } catch (error) {
        console.error('Error testing Sentry:', error);
        console.error('Make sure the sentry TypeScript dependencies are installed.');
        process.exit(1);
    }
}

main();

File: scripts/test_sentry.py
#!/usr/bin/env python3
"""
Standalone script to test Sentry integration.
"""
import os
import sys

# Add the parent directory to the path to import sentry module
sys.path.append(os.path.join(os.path.dirname(__file__), '..', 'sentry', 'py'))

try:
    from sentry import test_sentry_integration
    test_sentry_integration()
except ImportError as e:
    print(f"Error importing Sentry module: {e}")
    print("Make sure the sentry Python dependencies are installed.")
    sys.exit(1)
except Exception as e:
    print(f"Error testing Sentry: {e}")
    sys.exit(1)